tests/functional/modules/test_zos_archive_func.py

```diff
@@ -0,0 +1,935 @@
+    "format", [
+        "terse",
+        "xmit",
+        ])
+@pytest.mark.parametrize(
+    "data_set", [
+        dict(name=TEST_PS, dstype="seq"),
+        dict(name=TEST_PDS, dstype="pds"),
+        dict(name=TEST_PDS, dstype="pdse"),
+        ]
+)
+def test_mvs_archive_multiple_data_sets_with_missing(ansible_zos_module, format, data_set ):
+    try:
+        hosts = ansible_zos_module
+
+        target_ds_list = create_multiple_data_sets(ansible_zos_module=hosts,
+                                  base_name=data_set.get("name"),
+                                  n=3,
+                                  type=data_set.get("dstype"))
+        ds_to_write = target_ds_list
+        if data_set.get("dstype") in ["pds", "pdse"]:
+            target_member_list = []
+            for ds in target_ds_list:
+                target_member_list.extend(
+                    create_multiple_members(ansible_zos_module=hosts,
+                                        pds_name=ds.get("type"),
+                                        member_base_name="MEM",
+                                        n=3
+                    )
+                )
+            ds_to_write = target_member_list
+        # Write some content into src
+        test_line = "this is a test line"
+        for ds in ds_to_write:
+            hosts.all.shell(cmd="decho '{0}' \"{1}\"".format(test_line, ds.get("name")))
+
+        # Remove ds to make sure is missing
+        missing_ds = data_set.get("name")+"1"
+        hosts.all.zos_data_set(name=missing_ds, state="absent")
+        path_list = [ds.get("name") for ds in target_ds_list]
+
+        format_dict = dict(name=format, format_options=dict())
+        if format == "terse":
+            format_dict["format_options"].update(terse_pack="SPACK")
+        format_dict["format_options"].update(use_adrdssu=True)
+        archive_result = hosts.all.zos_archive(
+            path=path_list,
+            dest=MVS_DEST_ARCHIVE,
+            format=format_dict,
+        )
+        
+        # assert response is positive 
+        for result in archive_result.contacted.values():
+            print(result)
+            assert result.get("changed") is True
+            assert result.get("dest") == MVS_DEST_ARCHIVE
+            assert missing_ds in result.get("missing")
+            for ds in target_ds_list:
+                if ds.get("name") == missing_ds:
+                    assert ds.get("name") not in result.get("archived")
+                else:
+                    assert ds.get("name") in result.get("archived")
+            cmd_result = hosts.all.shell(cmd = "dls {0}.*".format(HLQ))
+            for c_result in cmd_result.contacted.values():
+                assert MVS_DEST_ARCHIVE in c_result.get("stdout")
+        
+    finally:
+        hosts.all.shell(cmd="drm {0}*".format(data_set.get("name")))
+        hosts.all.zos_data_set(name=MVS_DEST_ARCHIVE, state="absent")
+
+
+# Sent this test to the end bc of locking issues with other tests
```

Comment:
I suggest to move the declaration of programs to the top, just for consistency of all other tests we have.
Also delete print statements in 925, 812, 743, 676, 611, 549, 475, 399.